### DISCUSSION TITLE
Mac service not starting after restart

### DISCUSSION THREAD

**Discussion by ghost** [Ylianst/MeshCentral#6226] [Issues]
Hi,

I've installed the Mesh on my Mac and the service is not starting after Mac restart.
How can I configure it?
Thanks,

**Comment by si458** [comment]
How did u install the meshagent on ur mac?
as I believe I fixed this bug for the pkg method but not the terminal command line install methid

**Reply by ghost** [reply]
I used the commands
chmod +x meshagent
sudo ./meshagent -install

**Reply by si458** [reply]
yes this method doesnt work correctly still til new agents are made/released
the temp fix is here https://github.com/Ylianst/MeshAgent/issues/161#issuecomment-2002163963
basically mac wont load the LaunchDaemon/LaunchAgents if they have the same name!
so simply renaming one of them fixes the issue

**Reply by ghost** [reply]
I found the solution as I've written it somewhere and I'll share it here:
Change the Label 'meshagent' (or your app name) inside 'Library/LaunchAgents/meshagent.plist' to 'meshagent-launchagent' (or `yourappname-launchagent')and restart 

This works.

**Reply by ghost** [reply]
Now just the screen sharing problem exists.I've given permissions in the Mac for the agent but I only see a frozen Mac screen.
I get the message that I should configure the security options but the issue still remains.

**Reply by si458** [reply]
make sure u had added all 3 permission https://ylianst.github.io/MeshCentral/meshcentral/faq/#mac-clients
then restart ur mac

**Reply by ghost** [reply]
Can I do it with bash command somehow?

**Reply by si458** [reply]
no as these permissions need to be accepted by the end user of the device.
normally when you install it would pop up asking for permission, which in turn takes u to the correct pages to tick the box and allow
however due to mac bugs, they stopped working, but you can add them in manually urself

**Reply by ghost** [reply]
The documentation is very old and the UI is different.
There is no keyboard permission anymore just "screen records and audio capture" and "Drive permission".
I also see that a permission is given with the installation without a manual approvement.
I do see an issue when I need to give disable the permission and reenable to see the screen otherwise I just see the desktop without anything on it.
I tried to remove the entries and reenabled the destop to get the notification and in the permission section **I only see the meshagent** entry **and not the meshagent_osx64 entry**.
Where is the meshagent_osx64 is located so I can add it manually?
Thanks.

**Reply by si458** [reply]
Yes the screenshots are only for reference and they where took on my old macbook pro 10.16.

You can check here where apple have moved the settings too now https://support.apple.com/en-gb/guide/mac-help/mchl211c911f/14.0/mac/14.0

And it's Full Disk Access, Accessibility, Input Monitoring, Screen & System Audio Recording if i remember that needs enabled

The location of the app is normally located /usr/local/mesh_services/meshagent/meshagent
